flutter/lib/src/integrations/screenshot_integration.dart

```diff
@@ -1,29 +1,31 @@
 import 'dart:async';
 
 import 'package:sentry/sentry.dart';
-import 'package:sentry/sentry_private.dart';
-import '../screenshot/screenshot_attachment_processor.dart';
+import '../event_processor/screenshot_event_processor.dart';
 import '../sentry_flutter_options.dart';
 
-/// Adds [ScreenshotAttachmentProcessor] to options if [attachScreenshot] is true
+/// Adds [ScreenshotEventProcessor] to options event processors if [attachScreenshot] is true
 class ScreenshotIntegration implements Integration<SentryFlutterOptions> {
   SentryFlutterOptions? _options;
 
   @override
   FutureOr<void> call(Hub hub, SentryFlutterOptions options) {
     if (options.attachScreenshot) {
-      // ignore: invalid_use_of_internal_member
-      options.clientAttachmentProcessor =
-          ScreenshotAttachmentProcessor(options);
       _options = options;
-
+      options.addEventProcessor(ScreenshotEventProcessor(options));
       options.sdk.addIntegration('screenshotIntegration');
     }
   }
 
   @override
   FutureOr<void> close() {
-    // ignore: invalid_use_of_internal_member
-    _options?.clientAttachmentProcessor = SentryClientAttachmentProcessor();
+    final eventProcessors = _options?.eventProcessors
+            .where((element) => element.runtimeType == ScreenshotEventProcessor)
+            .toList() ??
+        [];
+
+    for (var eventProcessor in eventProcessors) {
+      _options?.removeEventProcessor(eventProcessor);
+    }
```

Comment:
You can cache the instance of `ScreenshotEventProcessor(options)` and if its not null, you remove it.
So you only remove the one created by the integration instead, no need for loop the processors list.